frontend/src/views/Quiz.vue

```diff
@@ -37,42 +54,80 @@ onBeforeMount(() => {
         });
 });
 
-const quiz = computed(() => store.state.currentQuiz);
+const quiz: ComputedRef<Quiz | null> = computed(() => store.state.currentQuiz);
 
-const currentQuestionIndex: Ref<number | null> = ref(null);
+const currentQuestion: Ref<QuizQuestion | null> = ref(null);
+const currentQuestionItem: Ref<Quizoot.QuestionItem | null> = ref(null);
+const currentQuestionNumber: Ref<number> = ref(0);
 
 const questionsFlow: ComputedRef = computed(() => {
     return {
         questionNumber:
-            currentQuestionIndex.value == null
-                ? 0
-                : currentQuestionIndex.value + 1,
-        isFirstQuestion: currentQuestionIndex.value == 0,
-        isLastQuestion:
-            currentQuestionIndex.value == quiz.value?.questions.length - 1,
+            currentQuestionItem.value == null ? 0 : currentQuestionNumber.value,
+        isFirstQuestion: currentQuestionItem.value?.prev_question_id == null,
+        isLastQuestion: currentQuestionItem.value?.next_question_id == null,
     };
 });
 
+function getQuizQuestionOfId(id: Quizoot.Question['id']) {
```

Comment:
I suggest we keep using the id. Don't forget this is the `Quiz` component. It should be responsible for fetching the quizzes only. Not questions.

I think we can just pass the id to the `Question` component or even put it in the url (like this: `/questions?id=1`). And then the `Question` compoennt will fetch the metadata of the question given the id.

Reply:
1. Doesn't that mean that a question will be fetched each time we click on a navigation button ? I thought that this was a behavior we were trying to avoid (less fetches as possible).<br><br>
2. I've already tried this approach, with the `const { data, error } = useFetch<T>(url, options);` syntax.
Then I faced an issue: the answers given in the `QuestionSpecs` were not "kept alive".

Anyway, I think I should work on it a bit more, there might surely be a way to do it as you are suggesting.

Reply:
1. Not really a big deal with caching 😁. The advantage of this approach is that it's cleaner. The Quiz component will focus on the essential
2. I'll take a look